CI: build fail on jruby re: bake-modernize dep "rugged"

A recent JRuby test run failed, due to some gems which needed a C ext ([rugged](https://github.com/libgit2/rugged)?), perhaps?

https://github.com/socketry/async-io/runs/1861635440

<details>

```
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--without-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/home/runner/.rubies/jruby-9.2.14.0/bin/jruby
	--with-sha1dc
	--without-sha1dc
	--use-system-libraries
	--with-git2-dir
	--without-git2-dir
	--with-git2-include
	--without-git2-include=${git2-dir}/include
	--with-git2-lib
	--without-git2-lib=${git2-dir}/lib
	--with-git2lib
	--without-git2lib

To see why this extension failed to compile, please check the mkmf.log which can
be found here:

/home/runner/.rubies/jruby-9.2.14.0/lib/ruby/gems/shared/extensions/universal-java-1.8/2.5.0/rugged-1.1.0/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in
/home/runner/.rubies/jruby-9.2.14.0/lib/ruby/gems/shared/gems/rugged-1.1.0 for
inspection.
Results logged to
/home/runner/.rubies/jruby-9.2.14.0/lib/ruby/gems/shared/extensions/universal-java-1.8/2.5.0/rugged-1.1.0/gem_make.out

An error occurred while installing rugged (1.1.0), and Bundler cannot continue.
Make sure that `gem install rugged -v '1.1.0' --source 'https://rubygems.org/'`
succeeds before bundling.

In gems.rb:
  bake-modernize was resolved to 0.4.5, which depends on
    rugged
Error: Process completed with exit code 5.
```

</details>

--- a/.github/workflows/development.yml
+++ b/.github/workflows/development.yml
@@ -17,7 +17,7 @@ jobs:
           - 2.5
           - 2.6
           - 2.7
-          - 3.0
+          - "3.0"
         
         experimental: [false]
         env: [""]
@@ -30,6 +30,7 @@ jobs:
           - os: ubuntu
             ruby: jruby
             experimental: true
+            env: "JRUBY_OPTS=-X+O"
           - os: ubuntu
             ruby: head
             experimental: true

--- a/.github/workflows/development.yml
+++ b/.github/workflows/development.yml
@@ -30,7 +30,7 @@ jobs:
           - os: ubuntu
             ruby: jruby
             experimental: true
-            env: "JRUBY_OPTS=-X+O"
+            env: 'JRUBY_OPTS="-X+O --debug"'
           - os: ubuntu
             ruby: head
             experimental: true

--- a/async-io.gemspec
+++ b/async-io.gemspec
@@ -16,13 +16,4 @@ Gem::Specification.new do |spec|
 	spec.required_ruby_version = ">= 2.5"
 	
 	spec.add_dependency "async", "~> 1.14"
-	
-	spec.add_development_dependency "async-container", "~> 0.15"
-	spec.add_development_dependency "async-rspec", "~> 1.10"
-	spec.add_development_dependency "bake"
-	spec.add_development_dependency "bake-bundler"
-	spec.add_development_dependency "bake-modernize"
-	spec.add_development_dependency "bundler"
-	spec.add_development_dependency "covered"
-	spec.add_development_dependency "rspec", "~> 3.0"
 end

--- a/gems.rb
+++ b/gems.rb
@@ -2,12 +2,22 @@
 
 source 'https://rubygems.org'
 
-# Specify your gem's dependencies in async-io.gemspec
 gemspec
+
+group :development do
+	gem "async-container", "~> 0.15"
+	gem "async-rspec", "~> 1.10"
+	gem "bake"
+	gem "bake-bundler"
+	gem "bake-modernize" unless defined?(JRUBY_VERSION)
+	gem "bundler"
+	gem "covered"
+	gem "rspec", "~> 3.0"
+end
 
 group :test do
 	gem 'benchmark-ips'
 	gem 'ruby-prof', platforms: :mri
-	
+
 	gem 'http'
 end

--- a/spec/async/io/shared_endpoint/server_spec.rb
+++ b/spec/async/io/shared_endpoint/server_spec.rb
@@ -67,6 +67,6 @@ RSpec.describe Async::Container::Forked, if: Process.respond_to?(:fork) do
 	it_behaves_like Async::IO::SharedEndpoint, described_class
 end
 
-RSpec.describe Async::Container::Threaded, if: RUBY_PLATFORM !~ /darwin/ do
+RSpec.describe Async::Container::Threaded, if: (RUBY_PLATFORM !~ /darwin/ && RUBY_ENGINE != "jruby") do
 	it_behaves_like Async::IO::SharedEndpoint, described_class
 end

